src/components/Toolbar/index.jsx

```diff
@@ -0,0 +1,121 @@
+/* global cozy */
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Menu, { Item } from 'components/Menu'
+import styles from './index.styl'
+
+import { withBreakpoints } from 'cozy-ui/react'
+
+const { BarRight } = cozy.bar
+
+class Toolbar extends Component {
+  render() {
+    const {
+      children,
+      moreMenuTitle,
+      moreMenuDisabled,
+      moreMenuClassName,
+      moreMenuButton,
+      breakpoints: { isMobile }
+    } = this.props
+
+    const actions = React.Children.toArray(children)
+    const highlightedActions = actions.filter(child => child.props.highlighted)
+    const secondaryActions = actions.filter(child => !child.props.highlighted)
+
+    const MoreMenu = (
+      <Menu
+        title={moreMenuTitle}
+        disabled={moreMenuDisabled}
+        className={moreMenuClassName}
+        button={moreMenuButton}
+      >
+        {highlightedActions.map(component =>
+          this.wrapInItem(
+            React.cloneElement(component, {
+              highlighted: false,
+              isSecondaryAction: false
+            })
+          )
+        )}
+        {secondaryActions.map(component =>
+          this.wrapInItem(
+            React.cloneElement(component, {
+              highlighted: false,
+              isSecondaryAction: true
+            })
+          )
+        )}
+      </Menu>
+    )
+
+    return (
+      <div role="toolbar">
+        {highlightedActions.map(component =>
+          React.cloneElement(component, {
+            highlighted: true,
+            isSecondaryAction: false
+          })
+        )}
+        {isMobile ? <BarRight>{MoreMenu}</BarRight> : MoreMenu}
+      </div>
+    )
+  }
+
+  wrapInItem(component) {
+    return <Item>{component}</Item>
+  }
+}
+
+Toolbar.propTypes = {
+  moreMenuTitle: PropTypes.string.isRequired,
+  moreMenuDisabled: PropTypes.bool,
+  moreMenuClassName: PropTypes.string,
+  moreMenuButton: PropTypes.element.isRequired,
+  children: PropTypes.node,
+  breakpoints: PropTypes.object
+}
+
+Toolbar.defaultProps = {
+  moreMenuDisabled: false,
+  moreMenuClassName: ''
+}
+
+export class ToolbarAction extends Component {
+  render() {
+    try {
+      const { children, visible, highlighted, isSecondaryAction } = this.props
+
+      if (!visible) return null
+
+      const child = React.cloneElement(React.Children.only(children), {
+        highlighted
+      })
```

Comment:
Why do you use both JSX and React API? I find it very confusing.

Reply:
I'm only using the React API to clone elements and manipulate `children`, things I can't do in JSX AFAIK.

Reply:
```jsx
const cloneComponent = BaseComponent => props => <BaseComponent {...props} />
```
seems to be almost the same as `React.cloneElement`, except in the JSX version, `ref` value attached to the `BaseComponent` will not be maintained. 

https://codesandbox.io/s/q9lv0j2k6